department-of-veterans-affairs/vets-website
# Add custom title A11y hack and confirmation modals

## Summary
On the Pension Benefits application(21P-527EZ), update the list loop patterns to pass A11y accessibility issue when rendering list loop data on the 'Review application' page. Add confirmation modals when removing a list loop item.

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify confirmation modals
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Start or continue the application
3. Go to the 'Military history' chapter
4. Continue to the 'Other service names' page
5. Select 'Yes' and continue to the 'Add other service names' page
6. Add multiple names
7. Verify that a confirmation modal is opened when a name is removed
8. Verify that all list loop items have a confirmation modal including on the 'Review application' page

## How to verify A11y issues
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Start or continue the application
3. Load maximal-data.json into the 'SIP' menu
4. Return to the 'Review application' page 
5. Expand all of the chapters to reveal the entire application
6. Open Axe in dev tools and scan the page
7. Confirm that there are no accessibility issues reported
8. Run cypress tests with `yarn cy:run --spec src/applications/pensions` and verify success for all three data sets


## Related issue(s)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

## Screenshots

### Desktop
| Confirmation modal |
| ------ |
|![Screenshot 2024-01-25 at 10 09 09 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/9075711f-e1c1-4384-91c5-584f39d62bf8)


## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
n/a

### Quality Assurance & Testing

### Error Handling


### Authentication
